### PR TITLE
[OM-73583]: Update go bindings for group dto protobuf

### PR DESCRIPTION
**Pull request by pallavidn**
NODE_POOL group type is introduced in server side SDK protobuf to represent node pools that exist in the kubernetes environments.
- Re-generated go bindings for the upated protobuf.
- Update to the group builder to build node pool groups by go probes

### CODE REVIEW

**Review by esara — APPROVED** [review]

**Comment by enlinxu** [comment]
The functionality change looks good to me. I am not sure about the protobuf generated files...
@marvelyue Can you please confirm?

**Comment by marvelyue** [comment]
@enlinxu , yes sure!

Hi @pallavidn , looks the generated file changes are too much. I tried to generate pb.proto files using all lastest proto files under opsmgr repo (including the new group type your introduced `NODE_POOL` in CommonDTO.proto) by this command:
```
protoc --go_out=import_path=proto:. *.proto
```
And the modified files are just:
```
	modified:   ActionExecution.pb.go
	modified:   CloudCostDTO.pb.go
	modified:   CommonCost.pb.go
	modified:   CommonDTO.pb.go
	modified:   Discovery.pb.go
	modified:   IdentityMetadata.pb.go
	modified:   MediationMessage.pb.go
	modified:   NonMarketEntityDTO.pb.go
	modified:   PlanExport.pb.go
	modified:   PricingDTO.pb.go
	modified:   ProfileDTO.pb.go
	modified:   SupplyChain.pb.go
	new file:   CostBilling.pb.go
```

And the changes of each `pb.go` file are not much, for example, in `CommonDTO.pb.go`, only couple of places get changed:
Screen Shot 2021-09-10 at 5 09 53 PM

I didn't see changes related to vendors either.

I'm using this protoc version:
```
$ protoc --version
libprotoc 3.11.4
```

What command did you use to generate pb.go files and which protoc version were you using? (maybe we can talk offline)

**Comment by pallavidn** [comment]
go bindings are generated using a newer version.

➜  proto-bindings protoc --version
protoc --version
libprotoc 3.17.3

File: go.mod [hunk 1]
```diff
@@ -7,8 +7,9 @@ require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/deckarep/golang-set v1.7.1
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	github.com/golang/protobuf v1.3.1
+	github.com/golang/protobuf v1.5.0
```
Comment (marvelyue) [inline go.mod:10]: Looks the latest golang/protobuf version is v1.5.2 based on https://github.com/golang/protobuf/releases. Shall we use the latest version here?

**Review by marvelyue — APPROVED** [review]
Talked with @ading1977 and @pallavidn. The generated file changes are because of a new protobuf version, which look to be fine.

Both `protoc-gen-go` and `protoc ` are the latest stable version:
```
// versions:
// 	protoc-gen-go v1.27.1
// 	protoc        v3.17.3
```

One minor comment on the golang/protobuf version in go.mod.

Overall the changes look good. And would be great if you can do a sanity test to see if entities from kubeturbo are properly discovered and sent to the platform (for example, cloud native entities are shown in supply chain and commodity values are presented properly in the "Capacity and Usage" widgets). (Please ignore this comment if you have done so)

And per discussion, would also be helpful to ask QE to run more regression tests on kubeturbo to make sure the update of protobuf won't cause unexpected behaviors.

**Pull request merged**